src/Simplify.elm

```diff
@@ -3860,7 +3866,18 @@ listFoldAnyDirectionChecks foldOperationName checkInfo =
                 numberBinaryOperationChecks { two = "+", list = "sum", identity = 0 }
 
             else if isBinaryOperation "*" checkInfo checkInfo.firstArg then
-                numberBinaryOperationChecks { two = "*", list = "product", identity = 1 }
+                if getUncomputedNumberValue initialArgument == Just 0 then
+                    [ Rule.errorWithFix
+                        { message = "The call to List." ++ foldOperationName ++ " (*) 0 will result in 0."
```

Comment:
Considering https://github.com/jfmengels/elm-review-simplify/issues/58, this is not really true, right?
```elm
List.foldl (*) 0 [0/0]
--> NaN
```

Reply:
Yes. Depending on what we decide on that issue, we'll have to do the same here